src/passport-saml/saml.ts

```diff
@@ -108,99 +108,80 @@ async function promiseWithNameID(nameid: Node): Promise<NameID> {
-     * - exact : Assertion context must exactly match a context in the list
-     * - minimum:  Assertion context must be at least as strong as a context in the list
-     * - maximum:  Assertion context must be no stronger than a context in the list
-     * - better:  Assertion context must be stronger than all contexts in the list
-     */
-    if (
-      !options.RACComparison ||
-      ["exact", "minimum", "maximum", "better"].indexOf(options.RACComparison) === -1
-    ) {
-      options.RACComparison = "exact";
-    }
+    const options = {
+      ...ctorOptions,
+      passive: ctorOptions.passive ?? false,
+      disableRequestedAuthnContext: ctorOptions.disableRequestedAuthnContext ?? false,
+      additionalParams: ctorOptions.additionalParams ?? {},
+      additionalAuthorizeParams: ctorOptions.additionalAuthorizeParams ?? {},
+      additionalLogoutParams: ctorOptions.additionalLogoutParams ?? {},
+      forceAuthn: ctorOptions.forceAuthn ?? false,
+      skipRequestCompression: ctorOptions.skipRequestCompression ?? false,
+      disableRequestAcsUrl: ctorOptions.disableRequestAcsUrl ?? false,
+      acceptedClockSkewMs: ctorOptions.acceptedClockSkewMs ?? 0,
+      path: ctorOptions.path ?? "/saml/consume",
+      host: ctorOptions.host ?? "localhost",
+      issuer: ctorOptions.issuer ?? "onelogin_saml",
+      identifierFormat:
+        ctorOptions.identifierFormat === undefined
+          ? "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress"
+          : ctorOptions.identifierFormat,
+      authnContext: ctorOptions.authnContext ?? [
+        "urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport",
+      ],
+      validateInResponseTo: ctorOptions.validateInResponseTo ?? false,
+      cert: assertRequired(ctorOptions.cert, "cert is required"),
+      requestIdExpirationPeriodMs: ctorOptions.requestIdExpirationPeriodMs ?? 28800000, // 8 hours
+      cacheProvider:
+        ctorOptions.cacheProvider ??
+        new InMemoryCacheProvider({
+          keyExpirationPeriodMs: ctorOptions.requestIdExpirationPeriodMs,
+        }),
+      logoutUrl: ctorOptions.logoutUrl ?? ctorOptions.entryPoint ?? "", // Default to Entry Point
+      signatureAlgorithm: ctorOptions.signatureAlgorithm ?? "sha1", // sha1, sha256, or sha512
+      authnRequestBinding: ctorOptions.authnRequestBinding ?? "HTTP-Redirect",
+
+      RacComparison: (() => {
+        /**
+         * List of possible values:
+         * - exact : Assertion context must exactly match a context in the list
+         * - minimum:  Assertion context must be at least as strong as a context in the list
+         * - maximum:  Assertion context must be no stronger than a context in the list
+         * - better:  Assertion context must be stronger than all contexts in the list
+         */
+        ctorOptions.RacComparison = ctorOptions.RacComparison ?? "exact";
+        if (!["exact", "minimum", "maximum", "better"].includes(ctorOptions.RacComparison)) {
```

Comment:
Why do the check in an IIFE ? I would find it clearer to have it after the object initialization

Reply:
The idea was to keep all the logic right together so there wouldn't be two places, even in the same ctor, to look for how a property ends up the way it is. However, I can see how IIFE would actually mutate the incoming obejct and make things slower. Done.